src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int type = random.nextInt(5);
+
+        String color = colorSupplier.getRandomColor();
+        switch (type) {
+            case 0:
+                double side = random.nextDouble() * 10 + 1;
+                return new Square(side, color);
+            case 1:
+                double length = random.nextDouble() * 10 + 1;
+                double width = random.nextDouble() * 10 + 1;

```

Comment:
this expression is used many times, extract it to private method with proper naming